docs/designers-developers/developers/themes/theme-json.md

```diff
@@ -0,0 +1,219 @@
+# Themes & Block Editor
+
+This is intended to document the current direction and work in progress about how themes can hook into the various sub-systems that the Block Editor provides.
+
+* Rationale
+* Specification
+* Current Status
+
+## Rationale
+
+The Block Editor surface API has evolved at different velocities, and it's now at a point where is showing some growing pains, specially in areas that affect themes. Examples of this are: the ability to [control the editor programmatically](https://make.wordpress.org/core/2020/01/23/controlling-the-block-editor/), or [a block style system](https://github.com/WordPress/gutenberg/issues/9534) that facilitates user, theme, and core style preferences.
+
+This describes the current efforts to consolidate the various APIs into a single point: a `experimental-theme.json` file.
+
+When this file is present and the Full Site Editing experiment is enabled, a few Block Editor mechanisms that are activated.
```

Comment:
Do we really need to limit the theme.json to FSE?

Reply:
ok, so there are two things here:

- The FSE experiment needs to be enabled for this to work. I liked that as it follows what the other experiments are doing. However, one alternative would be to enable this if there's an `experimental-theme.json`, independently of the FSE experiment status. Would that make more sense? I guess if a theme provides that file it knows what's doing, so it should be fine. I don't have a strong opinion on this.

- The stylesheet is only enqueued for front end & site editor. I wondered if we could open this to the other editors as well (post & widgets).

Reply:
> However, one alternative would be to enable this if there's an experimental-theme.json, independently of the FSE experiment status. Would that make more sense?

Yes, that was my thinking, I see value for this for regular themes too.

> The stylesheet is only enqueued for front end & site editor. I wondered if we could open this to the other editors as well (post & widgets).

Definitely, especially "post". What's the reasoning for loading this in the site editor page only?

Reply:
Draft PR to open this up https://github.com/WordPress/gutenberg/pull/22520

> What's the reasoning for loading this in the site editor page only?

Just being extra-cautious about which contexts had this loaded. With the current direction (enable features, etc), it makes more sense to open it up.